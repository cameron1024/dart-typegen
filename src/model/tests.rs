use super::*;

macro_rules! parse_snapshot {
    ($name:ident) => {
        #[test]
        fn $name() {
            let text = include_str!(crate::test_file!($name));
            let config = Library::parse_impl(None, text).unwrap();

            insta::assert_debug_snapshot!(config);
        }
    };
}

crate::all_test_files!(parse_snapshot);
