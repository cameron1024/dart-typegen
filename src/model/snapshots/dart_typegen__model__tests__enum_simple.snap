---
source: src/model/tests.rs
expression: config
---
Library {
    preamble: None,
    postamble: None,
    defaults: None,
    enums: [
        Enum {
            span: Span(
                0,
                0,
            ),
            name: SpannedScalar {
                value: "Color",
                span: SourceSpan {
                    offset: SourceOffset(
                        24,
                    ),
                    length: 7,
                },
            },
            docs: None,
            variants: [
                EnumVariant {
                    name: SpannedScalar {
                        value: "red",
                        span: SourceSpan {
                            offset: SourceOffset(
                                44,
                            ),
                            length: 5,
                        },
                    },
                    docs: None,
                    json_value: None,
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "green",
                        span: SourceSpan {
                            offset: SourceOffset(
                                60,
                            ),
                            length: 7,
                        },
                    },
                    docs: None,
                    json_value: None,
                },
                EnumVariant {
                    name: SpannedScalar {
                        value: "blue",
                        span: SourceSpan {
                            offset: SourceOffset(
                                78,
                            ),
                            length: 6,
                        },
                    },
                    docs: None,
                    json_value: None,
                },
            ],
        },
    ],
    classes: [],
    unions: [],
}
